src/server/scripts/Northrend/IcecrownCitadel/boss_the_lich_king.cpp

```diff
@@ -815,6 +815,7 @@ class boss_the_lich_king : public CreatureScript
                 {
                     me->GetMap()->SetZoneOverrideLight(AREA_ICECROWN_CITADEL, LIGHT_SNOWSTORM, 5000);
                     me->GetMap()->SetZoneWeather(AREA_ICECROWN_CITADEL, WEATHER_STATE_LIGHT_SNOW, 0.5f);
+                    summons.DespawnEntry(NPC_SHADOW_TRAP);
```

Comment:
@Demonid SpellHitTarget is called when spell hit a target right?
So it will call Despawn Shadow Trap alot of times no? (same to maps modifications)